packages/next/server/web/next-url.ts

```diff
@@ -102,17 +113,22 @@ export class NextURL {
 
     if (
       this[Internal].locale?.locale &&
-      i18n?.defaultLocale !== this[Internal].locale?.locale &&
-      !this.hasPathPrefix('/api')
+      ((i18n?.defaultLocale !== this[Internal].locale?.locale &&
+        !this.hasPathPrefix('/api')) ||
+        this[Internal].buildId)
     ) {
       pathname = `/${this[Internal].locale?.locale}${pathname}`
     }
 
-    if (this[Internal].basePath) {
-      pathname = `${this[Internal].basePath}${pathname}`
-    }
+    return this[Internal].buildId
+      ? `/_next/data/${this[Internal].buildId}${
```

Comment:
```suggestion
      ? `${this[Internal].basePath}/_next/data/${this[Internal].buildId}${
```
Wouldn't the `_next/data` URL also need to be prefixed with the `basePath`?

Reply:
Indeed! missed it, will also add some tests